### PR TITLE
Feature browse sources with limit and offset

### PR DESCRIPTION
**Pull request by pfeuffer**
### Your checklist for this pull request

_Put an `x` in the boxes that apply. You can also fill these out after creating the PR. If you're unsure about any of them, don't hesitate to ask. We're here to help! This is simply a reminder of what we are going to look for before merging your code._

- [x] PR is well described
- [x] Target branch is not master (in most cases develop should bet the target of choice) 
- [x] Code does not conflict with target branch
- [x] New code is covered with unit tests
- [x] CHANGELOG.md updated
- [ ] Documentation updated (only necessary for new features or changed behaviour)

### Checklist for branch merge request (not required for forks)

- [ ] Branch is green/blue on [Jenkins](https://oss.cloudogu.com/jenkins/)
- [ ] Quality Gate passed on [SonarQube](https://sonarcloud.io/organizations/scm-manager/projects)

### CODE REVIEW

File: scm-plugins/scm-hg-plugin/src/main/resources/sonia/scm/hg/ext/fileview.py [hunk 1]
```diff
@@ -271,14 +307,18 @@ def view(self, path = ""):
     ('d', 'disableLastCommit', False, 'disables last commit description and date'),
     ('s', 'disableSubRepositoryDetection', False, 'disables detection of sub repositories'),
     ('t', 'transport', False, 'format the output for command server'),
+    ('l', 'limit', 1000, 'limit the number of results'),
```
Comment (eheimbuch) [inline scm-plugins/scm-hg-plugin/src/main/resources/sonia/scm/hg/ext/fileview.py]: why is this limit set to 1000 instead 100?

File: scm-ui/ui-webapp/src/repos/sources/modules/sources.test.ts [hunk 2]
```diff
@@ -166,7 +191,7 @@ describe("sources fetch", () => {
       const actions = store.getActions();
       expect(actions[0].type).toBe(FETCH_SOURCES_PENDING);
       expect(actions[1].type).toBe(FETCH_SOURCES_FAILURE);
-      expect(actions[1].itemId).toBe("scm/core/_/");
+      expect(actions[1].itemId).toBe("scm/core/_//");
```
Comment (eheimbuch) [inline scm-ui/ui-webapp/src/repos/sources/modules/sources.test.ts]: i don't get this. why are you expecting two ending slashes?

Comment (pfeuffer) [reply scm-ui/ui-webapp/src/repos/sources/modules/sources.test.ts]: I've appended the hunk with a slash. This is the root path without an explicit hunk

File: scm-ui/ui-webapp/src/repos/sources/components/FileTree.tsx [hunk 3]
```diff
@@ -7,26 +7,40 @@ import styled from "styled-components";
...
-import { fetchSources, getFetchSourcesFailure, getSources, isFetchSourcesPending } from "../modules/sources";
+import {
+  fetchSources,
+  getFetchSourcesFailure,
+  getHunkCount,
+  getSources,
+  isFetchSourcesPending
+} from "../modules/sources";
 import FileTreeLeaf from "./FileTreeLeaf";
+import Button from "@scm-manager/ui-components/src/buttons/Button";
```
Comment (sdorra) [inline scm-ui/ui-webapp/src/repos/sources/components/FileTree.tsx]: do not import from source. Use "@scm-manager/ui-components".

**Pull request merged**